internal/pkg/v2/infrastructure/redis/device.go

```diff
@@ -143,9 +143,9 @@ func deleteDevice(conn redis.Conn, device models.Device) errors.EdgeX {
 	_ = conn.Send(DEL, storedKey)
 	_ = conn.Send(ZREM, DeviceCollection, storedKey)
 	_ = conn.Send(HDEL, DeviceCollectionName, device.Name)
-	_ = conn.Send(ZREM, fmt.Sprintf("%s:%s", DeviceCollectionServiceName, device.ServiceName), storedKey)
+	_ = conn.Send(ZREM, fmt.Sprintf("%s%s%s", DeviceCollectionServiceName, DBKeySeparator, device.ServiceName), storedKey)
```

Comment:
looks like a helper function would be good for the Sprintf to create the key. Something like:
```
CreateKey(collection string, keyName string)
```

Reply:
Agree with Lenny.  

SonarCloud also complains about 5 code smells, all of which are related to the duplication of `%s%s%s`

Moreover, per [benchmark experiment](https://pastebin.com/ukxUmAwa), `fmt.Sprintf` has the worst efficiency in go to concat simple strings.  Instead, I'd suggest to use `strings.Builder` to do the concatenation.  A sample code may look as shown below:
```
func CreateKey(targets ...string) string {
	var sb strings.Builder
	for _, s := range targets {
		sb.WriteString(s)
	}
	return sb.String()
}
``` 
And the line#146 could be revised to
```
_ = conn.Send(ZREM, CreateKey(DeviceCollectionServiceName, DBKeySeparator, device.ServiceName), storedKey)
```

Reply:
No problem, I also try strings.Join, it's more simple and efficient in our case.

Code for connecting strings:
```
func CreateKey(targets ...string) string {
	var sb strings.Builder
	for _, s := range targets {
		sb.WriteString(s)

	}
	return sb.String()
}

func CreateKeyByStringJoin(targets ...string) string {
	return strings.Join(targets,DBKeySeparator)
}
```

Code for running the benchmark testing:
```
func BenchmarkCreateKeyByStringBuilder(b *testing.B) {
	for i := 0; i < b.N; i++ {
		CreateKey(EventsCollectionDeviceName, DBKeySeparator,"TestDeviceName")
	}
}

func BenchmarkCreateKeyByStringJoin(b *testing.B) {
	for i := 0; i < b.N; i++ {
		CreateKeyByStringJoin(EventsCollectionDeviceName, "TestDeviceName")
	}
}

% go test -bench=. -run=none  
goos: darwin
goarch: amd64
pkg: github.com/edgexfoundry/edgex-go/internal/pkg/v2/infrastructure/redis
BenchmarkCreateKeyByStringBuilder-16            15157144                73.0 ns/op
BenchmarkCreateKeyByStringJoin-16               23520855                44.1 ns/op
```